src/Messages/FloorDataMessage.java

```diff
@@ -20,13 +24,30 @@
 	 * @param floorNumber indicating the floor of the message
 	 * @param direction to set direction of the message
 	 * @param destinationNumber destination floor of the message
+	 * @param fault 
 	 */
-	public FloorDataMessage(float timeStamp, int floorNumber, String direction, int destinationNumber) {
+	public FloorDataMessage(float timeStamp, int floorNumber, DirectionEnum direction, int destinationNumber, int fault) {
 		super();
 		this.timeStamp = timeStamp;
 		this.floorNumber = floorNumber;
 		this.direction = direction;
 		this.destinationNumber = destinationNumber;
+		
```

Comment:
We could also make new Messages (1 for transient faults and 1 for hard faults). A fault has no relation to the a floor message so it would be a good separation of concern. Also, transient faults need to give us an amount of time the fault extends for.